infra/concourse/build/developer-tools/build/scripts/execute_with_credentials.sh

```diff
@@ -0,0 +1,37 @@
+#! /bin/bash
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# Setup temp directory for credentials written to disk
+DELETE_AT_EXIT="$(mktemp -d)"
+export TMPDIR="${DELETE_AT_EXIT}"
+
+if [[ -z "${SERVICE_ACCOUNT_JSON}" ]]; then
```

Comment:
This looks duplicative. Can't this file just source setuo_environment and exec a function from there?

Reply:
The environment variable check is duplicated, and I did that to set expectations for anyone using this script (and to fail early before we got to `setup_environment.sh`). I don't have a problem removing it and deferring to the original

Reply:
Addressed in a5a6fce